### DISCUSSION TITLE
Interact with debug service of the iOS device

### DISCUSSION THREAD

**Discussion by icyflatline** [doronz88/pymobiledevice3#402] [Q&A]
Is it possible to interact with the debug service of the device via `pymobiledevice3` (similar to `idevicedebug` command of the `libimobiledevice` library)?

**Comment by doronz88** [comment]
All the functionality that can be accessed by mounting the DeveloperDiskImage, is wrapped in `developer` subcommand.

```shell
pymobiledevice3 developer debugserver
```

**Comment by icyflatline** [comment]
So `debugserver` subcommand is equivalent to `idevicedebugserverproxy` command in `libimobiledevice` library.  However, `libimobiledevice` also has an `idevicedebug` command that (at least based on my understanding) allows straightforward launching of the app  with the device debugserver session. Basically it is similar to launching an app with Xcode, while the xcode debug console output is shown in the `idevicedebug` stdout. 

I assume that there is no one-command equivalent in `pymobiledevice3` for now, correct?

**Reply by doronz88** [reply]
That's actually a very confusing and wrong `libimobiledevice` feature. That's not a `debugserver` feature at all, but DVT, meaning it's handled in `DTServiceHub`.

You can launch programs (use `--help` to view more options):

```shell
pymobiledevice3 developer dvt launch com.apple.mobilesafari
```

And also kill applications:

```shell
pymobiledevice3 developer dvt kill <pid>
```

Feel free to look arround the other amazing things DVT allows.

Also, the main project README refers to these commonly used examples:

https://github.com/doronz88/pymobiledevice3#example

**Reply by icyflatline** [reply]
I am more-or-less familiar with `developer dvt launch` command of `pymobiledevice3`. However, as far as I understand it is different from `idevicedebug` by the following:
- `developer dvt launch` doesn't interact with `debugserver` process of the device, it launches the app and returns the PID of the app on device. 
- `idevicedebug` launches an app, starts `debugserver` process on device and returns the debug console output of the launched app through the stdout.

Sorry If I am using the terminology incorrectly...

Whether they are both handled under-the-hood by `DTServiceHub` or not - I am sure you are much more familiar with this than me).

**Reply by doronz88** [reply]
you have one of three options:

1. launch with `dvt` and attach with `debugserver` using `lldb`
2. launch with `debugserver` and attach with `debugserver` using `lldb`
3. launch with `debugserver` and attach with `debugserver` and implement its protocol yourself in order to launch it yourself instead of the well-defined `lldb` client

**Reply by sachatt** [reply]
Is there any resolution to this ? With the recent change in iOS17 no longer being supported  by libimobiledevices I am trying to find a workaround to launch an add with debug service attached so I can view the outputs.
